src/app/datepicker/datepicker.directive.view.spec.ts

```diff
@@ -0,0 +1,473 @@
+import { MzDatepickerContainerComponent, MzDatepickerDirective } from './';
+
+describe('MzDatepickerDirective:view', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        FormsModule,
+      ],
+      declarations: [
+        MzDatepickerContainerComponent,
+        MzDatepickerDirective,
+        MzTestWrapperComponent,
+      ],
+    });
+  }));
+
+  describe('label', () => {
+    let nativeElement: any;
+
+    function label(): HTMLLabelElement {
+      return nativeElement.querySelector('label');
+    }
+
+    it('should be shown correctly when provided', async(() => {
+
+      buildComponent<any>(`
+        <mz-datepicker-container>
+          <input mz-datepicker
+            id="datepicker"
+            [label]="'label-x'">
+        </mz-datepicker-container>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(label().innerHTML).toBe('label-x');
+      });
+    }));
+
+    it('should be empty when omitted', async(() => {
+
+      buildComponent<any>(`
+        <mz-datepicker-container>
+          <input mz-datepicker
+            id="datepicker">
+        </mz-datepicker-container>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(label().innerHTML).toBe('');
+      });
+    }));
+  });
+
+  describe('placeholder', () => {
+    let nativeElement: any;
+
+    function input(): HTMLInputElement {
+      return nativeElement.querySelector('input')
+    }
+
+    it('should be shown correctly when provided', async(() => {
+
+      buildComponent<any>(`
+        <mz-datepicker-container>
+          <input mz-datepicker
+            id="datepicker"
+            [placeholder]="'placeholder-x'">
+        </mz-datepicker-container>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(input().getAttribute('placeholder')).toBe('placeholder-x');
+      });
+    }));
+
+    it('should not be shown correctly when omitted', async(() => {
+
+      buildComponent<any>(`
+        <mz-datepicker-container>
+          <input mz-datepicker
+            id="datepicker">
+        </mz-datepicker-container>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(input().getAttribute('placeholder')).toBeFalsy();
+      });
+    }));
+  });
+
+  describe('input', () => {
+    let nativeElement: any;
+
+    function input(): HTMLInputElement {
+      return nativeElement.querySelector('input');
+    }
+
+    it('should be disabled correctly when disabled attribute is true', async(() => {
+
+      buildComponent<any>(`
+        <mz-datepicker-container>
+          <input mz-datepicker
+            id="datepicker"
+            [disabled]="true">
```

Comment:
Should it use `disabled` instead of `[disabled]="true"` for this test?

Reply:
I added tests to validate both